core/src/lib.rs

```diff
@@ -331,4 +343,107 @@ mod tests {
+                "addressRegion": "ON",
+                "addressCountry": "Canada",
+                "postalCode": "M3B 1A2"
+            }),
+        );
+        cred_subject.insert(
+            "routingInfo".to_string(),
+            serde_json::json!({
+                "type":["RoutingInfo"],
+                "code": "GBDSC",
+                "value": "042962"
+            }),
+        );
+        cred_subject.insert(
+            "id".to_string(),
+            serde_json::json!("did:knox:z6Mk2cd21e9abe57fae7...31073da1b522790e63834fe17a4c2be"),
+        );
+
+        cred_subject.insert("givenName".to_string(), serde_json::json!("Alice"));
+        cred_subject.insert("familyName".to_string(), serde_json::json!("Smith"));
+        cred_subject.insert(
+            "iban".to_string(),
+            serde_json::json!("GB74GSLD04296280001319"),
+        );
+        cred_subject.insert("BIC11".to_string(), serde_json::json!("TDOMCATTTOR"));
+        let optional_properties = std::collections::HashMap::new();
+        let credential_id = "12345";
+
+        let credential = builder
+            .create_credential(
+                credential::CredentialType::BankAccount,
+                cred_subject,
+                optional_properties,
+                credential_id,
+                issuer,
+            )
+            .unwrap();
+
+        let value = serde_json::to_value(credential).unwrap().to_string();
+
+        println!("{value}");
+
+        // Create a "remote" document by parsing a file manually.
+        let input = json_ld::RemoteDocument::new(
+            // We use `IriBuf` as IRI type.
+            None,
+            // Optional content type.
+            None,
+            // Parse the file.
+            json_ld::syntax::Value::parse_str(
+                value.as_str(),
+                |span| span, // keep the source `Span` of each element as metadata.
+            )
+            .expect("Error creating remote document"),
+        );
+
+        // Use `NoLoader` as we won't need to load any remote document.
+        let mut loader = json_ld::NoLoader::<_, _>::new();
+
+        // Expand the "remote" document.
+        let expanded = aw!(input.expand(&mut loader));
+
+        match expanded {
+            Err(e) => {
+                println!("Error: {:?}", e);
+            }
+            Ok(expanded) => {
+                for object in expanded.into_value() {
+                    if let Some(id) = object.id() {
+                        let name = object
+                            .as_node()
+                            .unwrap()
+                            .get_any(&iri!("http://xmlns.com/foaf/0.1/name"))
+                            .unwrap()
+                            .as_str()
+                            .unwrap();
+
+                        println!("id: {id}");
+                        println!("name: {name}");
+                    }
+                }
+            }
+        }
+
+        // assert!(false);
```

Comment:
What assertions do we expect in this test, i.e. what are we testing in particular? Could you pls add?

Reply:
This test is being ignored, its meant to be followed back on once we actually are able to finish this issue completely. This PR is to include all of the other fixes made while putting work on this issue